### DISCUSSION TITLE
webRTC livestream - corporate networks peer-sharing setup best practices

### DISCUSSION THREAD

**Discussion by mistermarti** [ant-media/Ant-Media-Server#5652] [Q&A]
Dear community,

I have some questions about sending livestream via webRTC app, in order to deliver it to approx 2000 users worldwide, in different corporate networks with, for each, not enough Internet bandwidth to connect all of them in unicast.

I know there's some way to setup webRTC app that enables the players on the same LAN to peer-share the livestream between them and spare a lot of Internet bandwidth.

Do you know where I can find informations and best practices about this?
Can someones help or already know about that?

Thanks all for your help and have a nice day,

Nicolas

**Comment by tremzo** [comment]
Hi, @mistermarti let me find out internally how this would work. I am not too sure, it's the weekend so the response might be delayed.

**Reply by mistermarti** [reply]
Hi @timantmedia, thanks for your follow up. Let me know if you know about some ways to do that!

**Comment by USAMAWIZARD** [comment]
> I have some questions about sending live stream via webRTC app, in order to deliver it to approx 2000 users worldwide, in different corporate networks with, for each, not enough Internet bandwidth to connect all of them in unicast.

May be you can have  Ant Media Server in the cluster mode one on the  LAN and one on the cloud. Clients from the local network can fetch the streams  from the local Ant Media Server. Local AMS will be in Cluster mode with cloud one so it can fetch the streams from the cloud and distribute in the local network.

Or If you are planning to go with the mesh (for the local network ) , It will result in poor performance once the participants in the local network. Let me know how many players are you expecting on the local network.

> I know there's some way to setup webRTC app that enables the players on the same LAN to peer-share the livestream between them and spare a lot of Internet bandwidth.

When using the Media Server in the middle Like MCU architecture , I am not sure if its directly be possible because everything needs to flow from the media server , its totally possible when using p2p mesh architecture  , but scaling to 2000 user in a mesh Architecture  seems not possible.
